### PR TITLE
#19508 Personas working on GraphQL Page API

### LINKED ISSUE #19508: Make Rules work better in graphql
- url maps do not seem to work - either when firing on every request or when you add them to the detail page.
- Setting the persona in the query does not seem to work, I still get persona=null
- Also, it does not seem like we are accruing tags?  A visitor should accrue tags when they hit a detail page with tags in it.

```
{
  page(url: "/store/products/eagles-nest-outfitters-doublenest-hammock-patriot", fireRules:true, persona:"EcoEnthusiast") {
    viewAs{
      visitor{
        persona{
          keyTag
        }
        geo{
          continent
          country
          latitude
          longitude
        }
        tags{
          tag
        }
      }
    }
  }  
}
```

_Originally posted by @wezell in https://github.com/dotCMS/core/issues/19219#issuecomment-703885379_

### PR DESCRIPTION
**Pull request by dsilvam**
Changes introduced:
* Not throwing when geolocation can't be obtained.
* Accept a 'persona' parameter in the GraphQL Page API `page` field and use it to process the page. 
* Postman test for the above

### CODE REVIEW

**Review by victoralfaro-dotcms — APPROVED** [review]
LGTM

File: dotCMS/src/main/java/com/dotcms/visitor/business/VisitorAPIImpl.java [hunk 1]
```diff
@@ -105,6 +111,24 @@ public VisitorAPIImpl(LanguageWebAPI languageWebAPI, PersonaAPI personaAPI) {
...
+            persona = personaAPI.findPersonaByTag(personaID, user, true)
+                    .orElse(null);
+        }
+
+        return persona;
+    }
+
     public void removeVisitor(final HttpServletRequest request){
         DotPreconditions.checkNotNull(request, IllegalArgumentException.class, "Null Request");
         final HttpSession session = request.getSession(false);
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotcms/visitor/business/VisitorAPIImpl.java:134]: any test for the change in this class?

**Review by freddyucv — CHANGES_REQUESTED** [review]

File: dotCMS/src/main/java/com/dotcms/visitor/business/VisitorAPIImpl.java (hunk 1 above)
Comment (dsilvam) [reply dotCMS/src/main/java/com/dotcms/visitor/business/VisitorAPIImpl.java:134]: yes! new graphql test that covers these changes

**Review by freddyucv — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/visitor/business/VisitorAPIImpl.java (hunk 1 above)
Comment (dsilvam) [reply dotCMS/src/main/java/com/dotcms/visitor/business/VisitorAPIImpl.java:134]: Integration tests also added

**Review by freddyucv — APPROVED** [review]

**Review by erickgonzalez — APPROVED** [review]

**Pull request merged**